### ISSUE TITLE
Operator functions inside operators?

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/Modelica_LinearSystems2#18]
**Reported by per&#46;ostlund on 22 Nov 2012 12:16 UTC**
Is it legal for an operator to contain operator functions? I.e.
```
encapsulated operator '+'
  encapsulated function f
    ...
  end f;
end '+'
```
The spec says that an operator function may only be placed inside an operator record or a package inside an operator record, and although an operator is very similar to a package it is technically not a package. This is used in the Modelica_LinearSystems2 library, in DiscreteTransferFunction.'/', so I wonder if this is an error in that library or if the restriction on operator functions should include operators too.

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/909

**Comment by modelica-trac-importer** [comment]
**Comment by per&#46;ostlund on 22 Nov 2012 12:19 UTC**
Using encapsulated in the example was a copy/paste mistake, it has no relevance to the issue.

**Comment by modelica-trac-importer** [comment]
**Comment by choeger on 23 Nov 2012 09:45 UTC**
Maybe I am missing something here, but `f` is not a operator function, right? 

And as far as I see it, an operator function inside an operator makes no sense at all. An operator function is a short cut in case you only define one implementation of an operator for a given type. So you definitely need that type (the operator record).

**Comment by modelica-trac-importer** [comment]
**Comment by per&#x2e;ostlund on 23 Nov 2012 10:18 UTC**
Sorry, seems I messed up my example completely. It's supposed to be:
```
operator '+'
  operator function f
    ...
  end f;
end '+'
```

**Comment by modelica-trac-importer** [comment]
**Comment by choeger on 26 Nov 2012 08:41 UTC**
As I said, I assume this example to be useless, since `f` would either be the definition of `+` (and thus a plain function) or an (shortcut) operator on its own (and thus defining the operator `f` on `+`). 

In the examples you mentioned, I assume it is interpreted as a simple function? So the additional operator flag is simply ignored? I would vote to make this an error.

**Comment by modelica-trac-importer** [comment]
**Comment by hansolsson on 26 Nov 2012 12:54 UTC**
The example should according to 4.6 be handled as:
```
operator '+'
  operator f
     function f1
        ...
     end f1;
  end f;
end '+';
```
which seems illegal according to 4.6 (an operator may only contain functions; but here `operator '+'` contains the almost-package `operator f`).

An alternative would be to allow packages here in general; I don't see the benefit of that unless there are so many functions for one operator that we need to organize them better (seems very unlikely).

**Comment by modelica-trac-importer** [comment]
**Comment by sjoelund&#x2e;se on 4 Dec 2012 19:47 UTC**
Language group: Library is in error, so it should be updated.

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 22 Mar 2013 15:49 UTC**
Fixed in 32f7659bcc7073f33b0a83edb0512380c3dce46a by removing the wrong operator symbol at two places

**Comment by modelica-trac-importer** [comment]
**Modified by dietmarw on 22 Mar 2013 15:57 UTC**
